src/SelectPrompt.php

```diff
@@ -45,6 +45,15 @@ public function __construct(
             } else {
                 $this->highlighted = array_search($this->default, array_keys($this->options)) ?: 0;
             }
+
+            // If the default is not visible, scroll and center it.
+            // If it's near the end of the list, we just scroll to the end.
+            if($this->highlighted >= $this->scroll){
+                $optionsLeft = count($this->options) - $this->highlighted - 1;
+                $halfScroll = (int) floor(($this->scroll - 0.1) / 2);
```

Comment:
I'm curious what the 0.1 solves?

I'm guessing it's to just to make the selected item appear in the top half instead of the bottom half when `scroll` is an even number?

Reply:
While explaining why I did this, I encountered another add / even bug, so i rewrote this part (removed the confusing `- 0.1`) and made more automated tests on that part.

This is tricky because two things interacts with each other.

First, so we are on the same page, what's the problem? The problem is, when you are near the end of the list, you can't just center the default value because there is not enough items after it. 

For a scroll of 9, you end up with something like:

```
 │   ○ Portugal                                               │ │
 │   ○ South Africa                                           │ │
 │   ○ South Korea                                            │ │
 │   ○ Spain                                                  │ │
 │ › ● Switzerland                                            │ │
 │   ○ United Kingdom                                         │ │
 │   ○ United States                                          │ │
```

So we need an offset, here called `$endOffset`, to compensate the missing items and therefore to keep a list of 9 items:

```
 │   ○ Nigeria                                                │ │
 │   ○ New Zealand                                            │ │
 │   ○ Portugal                                               │ │
 │   ○ South Africa                                           │ │
 │   ○ South Korea                                            │ │
 │   ○ Spain                                                  │ │
 │ › ● Switzerland                                            │ │
 │   ○ United Kingdom                                         │ │
 │   ○ United States                                          │ │
```

The half scroll can't be a `float` so I floor it, 9 becomes 4.5, then 4. This means 4 items before and 4 items after the highlighted value, so 9 in total.

For an even scroll however, the floor does nothing, 10 becomes 5... so 5 items before, 5 items after... and the highlighted items in the center. 11 items in total, in instead of 10.

That's why we need to compensate for this highlighted item and decrement the `$endOffset`. This way, the highlighted item is taken into account and everything works.

You maybe noted the `$endOffset` value can end up being `-1` when we are not near the bottom of the list, which just put the highlighted value in the upper part (which is nicer than the bottom part, but that's just my opinion here).

These problems can quickly make you mad, I totally understand if this is still not 100% clear, especially when you have not worked on it. If you need more explanations, feel free to ask :)